blockchain/chain.go

```diff
@@ -934,9 +951,15 @@ func (b *BlockChain) disconnectBlock(node *blockNode, block, parent *dcrutil.Blo
 	subsidy := CalculateAddedSubsidy(block, parent)
 	newTotalSubsidy := curTotalSubsidy - subsidy
 
+	// Calcultate the next stake difficulty.
+	nextStakeDiff, err := b.CalcNextRequiredStakeDifficulty()
```

Comment:
This is wrong.  It would calculate the stake diff for the block after the one being disconnected.  You can just use the data that is already in `node.sbits` though since it's about to be disconnected and hence is the next stake difficulty as compared to the previous block for which the new state represents.